test/extended/baremetal/hosts.go

```diff
@@ -88,9 +140,48 @@ var _ = g.Describe("[sig-installer][Feature:baremetal] Baremetal platform should
 		o.Expect(hosts.Items).ToNot(o.BeEmpty())
 
 		for _, h := range hosts.Items {
-			expectStringField(h, "status.operationalStatus").To(o.BeEquivalentTo("OK"))
-			expectStringField(h, "status.provisioning.state").To(o.Or(o.BeEquivalentTo("provisioned"), o.BeEquivalentTo("externally provisioned")))
-			expectBoolField(h, "spec.online").To(o.BeTrue())
+			expectStringField(h, "baremetalhost", "status.operationalStatus").To(o.BeEquivalentTo("OK"))
+			expectStringField(h, "baremetalhost", "status.provisioning.state").To(o.Or(o.BeEquivalentTo("provisioned"), o.BeEquivalentTo("externally provisioned")))
+			expectBoolField(h, "baremetalhost", "spec.online").To(o.BeTrue())
+		}
+	})
+
+	g.It("have hostfirmwaresetting resources", func() {
+		skipIfNotBaremetal(oc)
+
+		dc := oc.AdminDynamicClient()
+
+		bmc := baremetalClient(dc)
+		hosts, err := bmc.List(context.Background(), v1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(hosts.Items).ToNot(o.BeEmpty())
+
+		hfsClient := hostfirmwaresettingsClient(dc)
+
+		for _, h := range hosts.Items {
+			hostName := getStringField(h, "baremetalhost", "metadata.name")
+
+			g.By("check that each baremetalhost has a corresponding hostfirmwaresettings")
```

Comment:
This message will be printed for each host, but it does not show any information about that. It could be improved by also adding the current hostname for more clarity

Reply:
Good point. Included the hostname in the debug output.